addons/pos_sale_margin/models/pos_order.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class PosOrderLine(models.Model):
+    _inherit = 'pos.order.line'
+
+    s_margin = fields.Float(compute='_compute_s_margin', store=True, help="It gives profitability by calculating the difference between the Unit Price and the cost price.")
+
+    @api.depends('price_subtotal', 'total_cost')
+    def _compute_s_margin(self):
+        for line in self:
+            line.s_margin = line.price_subtotal - line.total_cost
```

Comment:
There is already a margin field for `pos.order.line` in point_of_sale though?

Reply:
Yes, but this is a non-stored computed field so it was not possible to use it in a SQL query

Reply:
you can do the same thing as in `pos_order_report`:
```
SUM(l.price_subtotal - l.total_cost / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) AS margin
```

This was suggested by fp because he didn't want the margin the be stored cuz we already have too many fields and margin is just a substraction which is really fast in SQL as well

Reply:
You can do it the same as in `pos_order_report.py`: 
`SUM(l.price_subtotal - l.total_cost / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) AS margin`
This was suggested by FP cuz we already had too many fields and a substraction is a fast operation in SQL